upstreams/container.go

```diff
@@ -0,0 +1,88 @@
+/*
+Copyright 2021 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package upstreams
+
+import (
+	"sort"
+	"strings"
+
+	"github.com/blang/semver"
+	"github.com/pkg/errors"
+	log "github.com/sirupsen/logrus"
+
+	"sigs.k8s.io/zeitgeist/pkg/container"
+)
+
+// Container upstream representation
+type Container struct {
+	UpstreamBase `mapstructure:",squash"`
+	// Container URL, e.g. gcr.io/k8s-staging-kubernetes/conformance
+	URL string
+	// Optional: semver constraints, e.g. < 2.0.0
+	// Will have no effect if the dependency does not follow Semver
+	Constraints string
+}
+
+// LatestVersion returns the latest tag  for the given repository
+// (depending on the Constraints if set).
+func (upstream Container) LatestVersion() (string, error) { // nolint:gocritic
```

Comment:
Ah, I see that LatestVersion isn't something you can really control the name of (at least for the exposed method), but my comment re: naming confusion still applies to the private functions, IMO.

Reply:
maybe we can refactor that in the future